crates/driver/src/infra/blockchain/gas.rs

```diff
@@ -81,24 +83,25 @@ impl GasPriceEstimator {
         self.gas
             .estimate()
             .await
-            .map(|mut estimate| {
-                let estimate = match self.additional_tip {
-                    Some((max_additional_tip, additional_tip_percentage)) => {
-                        let additional_tip = max_additional_tip
+            .map(|estimate| {
+                let additional_tip = self
+                    .additional_tip
+                    .map(|(max_additional_tip, additional_tip_percentage)| {
+                        max_additional_tip
                             .to_f64_lossy()
-                            .min(estimate.max_fee_per_gas * additional_tip_percentage);
-                        estimate.max_fee_per_gas += additional_tip;
-                        estimate.max_priority_fee_per_gas += additional_tip;
-                        estimate
-                    }
-                    None => estimate,
-                };
+                            .min(estimate.max_fee_per_gas * additional_tip_percentage)
+                    })
+                    .map(eth::U256::from_f64_lossy)
+                    .unwrap_or(eth::U256::zero());
```

Comment:
This section is quite noisy. It could be made simpler by turning `self.additional_tip` into a non-optional value. In cases where it's currently optional we could just make the relative and the max `0`. This could then be used to have following code here:
```rust
                let (max, percentage) = self.additional_tip;
                let additional_tip = max
                    .to_f64_lossy()
                    .min(estimate.max_fee_per_gas * percentage);
```

Alternatively we could replace the type alias `AdditionalTip` with a proper struct and define a helper function on this.

Reply:
Updated.